src/charm.py

```diff
@@ -1,53 +1,45 @@
 #!/usr/bin/env python3
 
 import logging
-from pathlib import Path
 from random import choices
 from string import ascii_uppercase, digits
-from typing import Optional
+
 from ops.charm import CharmBase
+from ops.framework import StoredState
 from ops.main import main
 from ops.model import ActiveStatus, MaintenanceStatus
 
-from oci_image import OCIImageResource, OCIImageResourceError
 from charms.minio.v0.minio_interface import MinioProvide
+from oci_image import OCIImageResource, OCIImageResourceError
 
 log = logging.getLogger()
 
 
-def get_or_set(name: str, *, configured: Optional[str], default: str) -> str:
-    if configured:
-        Path(f"/run/{name}").write_text(configured)
-        return configured
-
-    try:
-        path = Path(f"/run/{name}")
-        return path.read_text()
-    except FileNotFoundError:
-        path.write_text(default)
-        return default
-
-
 def gen_pass() -> str:
     return "".join(choices(ascii_uppercase + digits, k=30))
 
 
 class MinioCharm(CharmBase):
+    _stored = StoredState()
+
     def __init__(self, *args):
+        if not self.model.unit.is_leader():
+            log.info("Not a leader, skipping set_pod_spec")
+            self.model.unit.status = ActiveStatus()
+            return
+
         super().__init__(*args)
```

Comment:
Ah, it seems the `super()` does have to come first.

```suggestion
        super().__init__(*args)
        if not self.model.unit.is_leader():
            log.info("Not a leader, skipping set_pod_spec")
            self.model.unit.status = ActiveStatus()
            return
```